### PR TITLE
feat(MPTv2): Adds ops for versioned v2 MPTs.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java [hunk 1]
```diff
@@ -84,15 +84,26 @@ public TaskResult execute(Stage stage) {
...
+        (String) stage.getContext().get("id"),
+        null,
+        (Map<String, Object>) stage.decodeBase64("/pipelineTemplate", Map.class));
+    } else {
+      response = front50Service.updateV2PipelineTemplate(
+        (String) stage.getContext().get("id"),
+        version,
+        (Map<String, Object>) stage.decodeBase64("/pipelineTemplate", Map.class));
+    }
+
```
Comment (lwander) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java]: Extra newline?

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/CreateV2PipelineTemplateTask.java [hunk 2]
```diff
@@ -66,15 +67,27 @@ public TaskResult execute(Stage stage) {
...
     validate(pipelineTemplate);
 
-    Response response = front50Service.saveV2PipelineTemplate((Map<String, Object>) stage.decodeBase64(
-      "/pipelineTemplate",
-      Map.class,
-      pipelineTemplateObjectMapper
-    ));
+    Response response;
+    String version = (String) stage.getContext().get("version");
+    if (StringUtils.isEmpty(version)) {
```
Comment (lwander) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/CreateV2PipelineTemplateTask.java]: I don't think you need this `if`, can't you just call saveV2PipelineTemplate with an empty version?

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java [hunk 3]
```diff
@@ -84,15 +84,26 @@ public TaskResult execute(Stage stage) {
...
     validate(pipelineTemplate);
 
-    // TODO(jacobkiefer): move this to an update call to front50.
-    Response response = front50Service.updateV2PipelineTemplate(
-      (String) stage.getContext().get("id"),
-      (Map<String, Object>) stage.decodeBase64("/pipelineTemplate", Map.class)
-    );
+    Response response;
+    String version = (String) stage.getContext().get("version");
+    if (StringUtils.isEmpty(version)) {
```
Comment (lwander) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java]: Same deal here with this conditional as above

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/CreateV2PipelineTemplateTask.java (hunk 2 above)
Comment (jtk54) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/CreateV2PipelineTemplateTask.java]: Yep that was a relic from the first try as a path parameter.

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java (hunk 1 above)
Comment (jtk54) [reply orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/v2/UpdateV2PipelineTemplateTask.java]: Yep.

**Comment by jtk54** [comment]
Depends on https://github.com/spinnaker/front50/pull/429 and https://github.com/spinnaker/gate/pull/708.

**Pull request merged**